src/main/java/core/basesyntax/impl/SupplyHandlerImpl.java

```diff
@@ -0,0 +1,16 @@
+package core.basesyntax.impl;
+
+import dao.FruitDao;
+import dao.FruitDaoImpl;
+import model.FruitTransaction;
+import strategy.OperationHandler;
+
+public class SupplyHandlerImpl implements OperationHandler {
+    private FruitDao fruitDao = new FruitDaoImpl();
```

Comment:
It's not a good practice to directly instantiate a `FruitDaoImpl` inside the `SupplyHandlerImpl` class. This violates the Dependency Inversion Principle, one of the SOLID principles. Instead, consider passing a `FruitDao` instance to the `SupplyHandlerImpl` through its constructor. This will make your code more testable and flexible.